packages/core/chrome/core-chrome-browser/src/contracts.ts

```diff
@@ -161,4 +162,26 @@ export interface ChromeStart {
    * Get an observable of the current style type of the chrome.
    */
   getChromeStyle$(): Observable<ChromeStyle>;
+  /**
+   * Configuration for serverless projects
+   */
+  project: {
+    /**
+     * Sets the project navigation config to be used for rendering project navigation.
+     * It is used for default project sidenav, project breadcrumbs, tracking active deep link.
+     * @param projectNavigation The project navigation config
+     *
+     * @remarks Has no effect if the chrome style is not `project`.
+     */
+    setNavigation(projectNavigation: ChromeProjectNavigation): void;
+
+    /**
+     * Set custom project sidenav component to be used instead of the default project sidenav.
+     * @param getter A function returning a CustomNavigationComponent.
+     * This component will receive Chrome navigation state as props (not yet implemented)
+     *
+     * @remarks Has no effect if the chrome style is not `project`.
+     */
+    setNavigationComponent(getter: () => CustomNavigationComponent): void;
```

Comment:
I was thinking we should call this `setSideNavComponent`: 

The idea is that setNavigation sets a configuration for navigation overall. Navigation includes sidenav, breadcrumbs, global search, etc. 

but here we specifically set a custom component just instead of a sidenav piece. 

WDYT?

Reply:
@Dosant maybe `s/setSideNavComponent/setNavigationComponent/`?

Reply:
Ok makes sense, I like `setSideNavComponent `. 👍